daemon/src/engine/server/CryptoChallege.h

```diff
@@ -0,0 +1,633 @@
+/*
```

Comment:
Library+Algorithm is to select an algorithm by name at runtime.

Once a secure rcon commands arrives, these steps are needed:
* Base64 decoding
* Key generation
* Decryption
* Hex Decoding
I'm not a fan of exceptions myself but I feel these many checks will be quite error-prone and might result in code duplication.

What trivial methods are you referring to? The functions that return constants are just placeholders until we decide where to get those value from (Cvar, constants or whatever).